src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -244,13 +205,46 @@ public JsonWebKey supplyKey(
 
         private final Map<String, Long> scopeBitsByName = new CopyOnWriteHashMap<>();
 
-        private long realmBit;
+        private final long realmId;
         private long nextScopeBitShift;
 
         private OAuthRealm(
-            long realmBit)
+            long realmId)
+        {
+            this.realmId = realmId;
+        }
```

Comment:
Consider adding `realmName` to constructor and also adding a `toString()` override to assist with debugging.